components/server/src/setup.rs

```diff
@@ -303,15 +303,29 @@ pub fn overwrite_config_with_cmd_args(config: &mut TikvConfig, matches: &ArgMatc
     }
 }
 
-#[allow(dead_code)]
 pub fn validate_and_persist_config(config: &mut TikvConfig, persist: bool) {
+    // Check current critical configurations with last time, if there are some
+    // changes, user must guarantee relevant works have been done.
+    let mut last_cfg = get_last_config(&config.storage.data_dir);
+    if let Some(last_cfg) = &mut last_cfg {
+        last_cfg.compatible_adjust();
+        if let Err(e) = last_cfg.validate() {
```

Comment:
if validate() returns error, should it be ignored and not used by optional_default_cfg_adjust_with?
Or if we don't care,  then no need to call compatible_adjust() and validate at all.

Reply:
check_critical_config needs `compatible_adjust()` and `validate()` is called, otherwise the config difference is not accurate